### PR TITLE
Convert <code> sections into cgalExamples

### PR DESCRIPTION
**Pull request by afabri**
This  at the same time fixes bugs as the examples where not 100 correct.

I have merged it into integration.

### CODE REVIEW

**Comment by afabri** [comment]
The doc testsuite is green
https://cgal.geometryfactory.com/CGAL/Members/Manual_doxygen_test/CGAL-4.7-Ic-148/output/Kernel_23/group__intersection__linear__grp.html

The Kernel_2 example testsuite is RED :<   => we need one more round.

**Comment by afabri** [comment]
The Kernel_2 example testsuite is YELLOW :< => we need one more round.

**Comment by sloriot** [comment]
[Kernel_23 tests](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-3.shtml) look good for now

File: Intersections_2/include/CGAL/Intersection_traits.h [hunk 1]
```diff
@@ -165,7 +165,7 @@ namespace internal {
   inline
   CGAL::Object intersection_return() { return CGAL::Object(); }
 #else
-  #if defined(CGAL_CFG_NO_CPP0X_RVALUE_REFERENCE)
+#if defined(CGAL_CFG_NO_CPP0X_RVALUE_REFERENCE)
```
Comment (lrineau) [inline Intersections_2/include/CGAL/Intersection_traits.h]: In my opinion, the correct indentation should have been:
```C++
#  if defined(CGAL_CFG_NO_CPP0X_RVALUE_REFERENCE)
```

**Pull request merged**